### PR TITLE
[docs] Add step to ensure sync_dist is adding to logging when multi-gpu enabled

### PR DESCRIPTION
**Pull request by SeanNaren**
What does this PR do?

Clears up a bunch of cases where `sync_dist` is required to ensure validation/test step results are synced across processes. 

## Before submitting
- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [x] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests? 
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:  

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified; _Bugfixes should be including in bug-fix release milestones (m.f.X) and features should be included in (m.X.b) releases._
 

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: docs/source/multi_gpu.rst [hunk 1]
```diff
@@ -103,6 +103,33 @@ Lightning adds the correct samplers when needed, so no need to explicitly add sa
 
 .. note:: For iterable datasets, we don't do this automatically.
 
+
+Synchronize validation and test logging
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+When using distributed, we have to ensure that the validation and test step logging calls are synchronized across processes.
+This is done by adding `sync_dist=True` to all `self.log` calls in the validation and test step.
+Ensures that each GPU worker has the same behaviour when tracking model checkpoints This is important for later downstream tasks, such as testing the best checkpoint across all workers.
```
Comment (SkafteNicki) [inline docs/source/multi_gpu.rst:112]:
```suggestion
This ensures that each GPU worker has the same behaviour when tracking model checkpoints, which is important for later downstream tasks such as testing the best checkpoint across all workers.
```

File: docs/source/multi_gpu.rst [hunk 2]
```diff
@@ -103,6 +103,33 @@ Lightning adds the correct samplers when needed, so no need to explicitly add sa
 
 .. note:: For iterable datasets, we don't do this automatically.
 
+
+Synchronize validation and test logging
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+When using distributed, we have to ensure that the validation and test step logging calls are synchronized across processes.
```
Comment (SkafteNicki) [inline docs/source/multi_gpu.rst:110]:
```suggestion
When running in distributed mode, we have to ensure that the validation and test step logging calls are synchronized across processes.
```

File: docs/source/multi_gpu.rst [hunk 3]
```diff
@@ -103,6 +103,33 @@ Lightning adds the correct samplers when needed, so no need to explicitly add sa
...
+This is done by adding `sync_dist=True` to all `self.log` calls in the validation and test step.
+Ensures that each GPU worker has the same behaviour when tracking model checkpoints This is important for later downstream tasks, such as testing the best checkpoint across all workers.
+
+Note if you use any built in metrics or custom metrics that use the ``pytorch_lightning.metrics`` API, these do not need to be updated and are handled for you.
```
Comment (SkafteNicki) [inline docs/source/multi_gpu.rst:114]:
```suggestion
Note if you use any built in metrics or custom metrics that use the ``pytorch_lightning.metrics`` API, these do not need to be updated and are automatically handled for you.
```

**Review by SkafteNicki — APPROVED** [review]
LGTM, just smaller comments

**Review by ananyahjha93 — APPROVED** [review]

File: docs/source/multi_gpu.rst [hunk 4]
```diff
@@ -103,6 +103,33 @@ Lightning adds the correct samplers when needed, so no need to explicitly add sa
...
 
+
+Synchronize validation and test logging
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+When running in distributed mode, we have to ensure that the validation and test step logging calls are synchronized across processes.
+This is done by adding `sync_dist=True` to all `self.log` calls in the validation and test step.
+This ensures that each GPU worker has the same behaviour when tracking model checkpoints, which is important for later downstream tasks such as testing the best checkpoint across all workers.
+
+Note if you use any built in metrics or custom metrics that use the ``pytorch_lightning.metrics`` API, these do not need to be updated and are automatically handled for you.
```
Comment (awaelchli) [inline docs/source/multi_gpu.rst:114]:
```suggestion
Note if you use any built in metrics or custom metrics that use the :ref:`Metrics API <metrics>`, these do not need to be updated and are automatically handled for you.
```

**Review by awaelchli — APPROVED** [review]

**Review by teddykoker — APPROVED** [review]

**Pull request merged**